src/middleware/request-id/request-id.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @module
+ * Request ID Middleware for Hono.
+ */
+
+import type { MiddlewareHandler } from '../../types'
+
+export type RequestIDVariables = {
+  requestID: string
+}
+
+export type RequesIDOptions = {
+  limitLength?: number
+  headerName?: string
+  generator?: () => string
```

Comment:
What about passing the context to the `generator`?

```ts
generator?: (c:Context) => string
```

Reply:
It looks nice.
This will allow users to handle request ID as they like.